### PR TITLE
Exit status consistency.Decomposed return coded and signal info.

### PR DESCRIPTION
**Pull request by nicfit**
Fixes issue #509

### CODE REVIEW

**Comment by scopatz** [comment]
Looks like there are merge conflicts @nicfit. Could you please take care of these.

**Comment by nicfit** [comment]
Good to go.

File: xonsh/proc.py [hunk 1]
```diff
@@ -418,3 +434,14 @@ def stdout(self):
         else:
             self._stdout = self._tpty.buffer
         return self._stdout
+
+
+def _wcode_to_popen(code):
+    '''Converts os.wait return code into Popen format.'''
```
Comment (scopatz) [inline xonsh/proc.py]: Can you convert the docstring to triple double quotes rather than triple single quote, please?

**Comment by nicfit** [comment]
Fixed

**Comment by scopatz** [comment]
Thanks a million for tracking all of this stuff down @nicfit! It is hugely valuable.

**Comment by nicfit** [comment]
Your welco

**Pull request merged**